core/jvm/src/main/scala/cats/effect/IOApp.scala

```diff
@@ -314,6 +314,32 @@ trait IOApp {
   protected def onCpuStarvationWarn(metrics: CpuStarvationWarningMetrics): IO[Unit] =
     CpuStarvationCheck.logWarning(metrics)
 
+  /**
+   * Defines what to do when IOApp detects that `main` is being invoked on a `Thread` which
+   * isn't the main process thread. This condition can happen when we are running inside of an
+   * `sbt run` with `fork := false`
+   */
+  private def onNonMainThreadDetected(): Unit = {
+    val shouldPrint =
+      Option(System.getProperty("cats.effect.warnOnNonMainThreadDetected"))
+        .map(_.equalsIgnoreCase("true"))
+        .getOrElse(true)
+    if (shouldPrint)
+      System
+        .err
+        .println(
+          """|Warning: IOApp `main` is running on a thread other than the main thread.
```

Comment:
```suggestion
          """|[WARNING] IOApp `main` is running on a thread other than the main thread.
```

We use this convention in the starvation checker